### ISSUE TITLE
Installed location

### ISSUE THREAD

**Issue by JoeKuhns** [Aluxian/electron-windows-installer#11]
Is it possible to change the installed location?  Right now we are having trouble running Windows Application Verification as the app doesn't install into Program Files.  Thanks!  love your work so far.

**Comment by aluxian** [comment]
It's not possible to change the install location, that's a design decision taken by Squirrel.Windows:

> Squirrel will never offer this choice, sorry!
> 
> https://github.com/atom/atom/issues/7095#issuecomment-109125522

See the entire discussion about it here: https://github.com/atom/atom/issues/7095
Squirrel.Windows's philosophy: https://github.com/squirrel/squirrel.windows#what-do-we-want

I think your options are to:
- create the installer normally and run it with `Setup.exe --machine` (or wrap Setup.exe together with a script that runs `Setup.exe --machine`)
- create the installer as an MSI (still, I'm not sure this will install where you want it to...)

---

Just curious:

What's Windows Application Verification? What are the advantages of it? Is it only for publishing in Windows Store? Links are welcome.

**Comment by JoeKuhns** [comment]
Okay, I'll play around.  Think their is any way to make --machine the default? Because that is exactly what I need.

<hr>

Windows Application Verification helps your reputation so you users down't see the ridculous SmartScreen warnings when they go to install your app.  While we have signed our app, we have no rep, so our users still see the warning.  

Here is a link to the verification tool.
https://www.microsoft.com/en-us/download/details.aspx?id=20028

Here is a link to a discussion on SO about the advantages...
http://stackoverflow.com/questions/12311203/how-to-pass-the-smart-screen-on-win8-when-install-a-signed-application

Through this tool you can create an XML file to upload to the SmartScreen review team, which will help further your rep. And actually get ride of the SmartScreen warning.

**Comment by aluxian** [comment]
I don't know how to make that the default... I was thinking you could write a batch script to run `setup.exe --machine` and then pack that script with setup.exe into a single exe... but I don't know if that'll work.

This might help:
- http://stackoverflow.com/questions/29453254/how-do-merge-two-exe-files-with-command-prompt-c
- https://www.google.com/search?q=how+to+merge+batch+script+with+exe

---

Oh, I remember reading about it... and I think I tried something, but some Microsoft pages didn't work so I gave up. I'll read more about it, thanks!

**Comment by aluxian** [comment]
I remembered, it was the Windows Logo program I tried. And by "try", I mean that I tried to access its page but it didn't work. Is the Windows Logo program still something developers want to get into to display that badge on their website? Or was that something specific to the Windows 7 years...

**Comment by JoeKuhns** [comment]
I don't know about the windows logo program as I live in a mac world.  But I wrote an app for a client and the end users are not the brightest, so if I can pass the windows app verifier and get rid of the SmartScreen warning it will make life easier for everyone.  My other option is just wait and as more people install the app, the certificate reputation will increase and the SmartScreen warning will go away on its own.  I'm still gonna see if I can hack my way around and make --machine the default.

**Comment by aluxian** [comment]
I read the SO thread you linked to and found Windows App Certification Kit. I'm running it right now to see how my apps perform.

> I live in a mac world

Me too... 😎

> just wait and as more people install the app, the certificate reputation will increase

Yeah, that's what I did...

**Comment by JoeKuhns** [comment]
Mine fails pretty quickly as it immediately doesn't like that is not in Program Files or x86...  But the scanner is sloooowww...

**Comment by aluxian** [comment]
I started WACK 5-10 minutes ago and it's on step "program inventory". I haven't tried the other verifier yet. Yeah, it's slow...

**Comment by aluxian** [comment]
Is this what you got? Or did you get this?

<img width="793" alt="screen shot 2016-06-14 at 17 24 03" src="https://cloud.githubusercontent.com/assets/1418870/16046408/067e3ba8-3255-11e6-85ca-041b5ff9a1fb.png">

**Comment by aluxian** [comment]
https://github.com/Squirrel/Squirrel.Windows/issues/452

**Comment by JoeKuhns** [comment]
Yup.  From what I read, that is because the only place microsoft thinks you should install an app is program files.

**Comment by aluxian** [comment]
Ahaha, great. It stopped working while saving the report:

<img width="712" alt="screen shot 2016-06-14 at 17 32 37" src="https://cloud.githubusercontent.com/assets/1418870/16046677/02dac24a-3256-11e6-948f-16fa183ad06f.png">

Whatever... 

>  that is because the only place microsoft thinks you should install an app is program files.

That means no certification for my apps, then...

**Comment by JoeKuhns** [comment]
Maybe no certification for mine too....  But I'll keep trying (i get paid by the hour).  I'll post here if I find any workaround.

**Comment by TheReincarnator** [comment]
Squirrel's choice is a bad one. IMHO, it puts computers at risk. Users should be aware of software getting installed on their PCs, so there should always be a head-up UAC message. I'd even wish that Microsoft forbids executable files in the User's directory altogether.
